### PR TITLE
Fixed default TRUST_DOMAIN to hostname of the first collector without port

### PR DESCRIPTION
**Pull request by mambelli**
Adding the possibility to set TRUST_DOMAIN via the attr TRUST_DOMAIN

By default TRUST_DOMAIN is GLIDEIN_Collector and if that is empty, CCB_ADDRESS.
Now if TRUST_DOMAIN is set it will be checked and used first.

And fixed TRUST_DOMAIN value: now it is the first collector (with port and synful string).

Considering only the first collector in the list (CE collectors could be added to the glidein). Secondary collectors could have a different port and synful string, but it is OK because the Glidein TRUST_DOMAIN does not have to be the same as the collector.
Both shell script and python code (the one to evaluate the issuer of the token) have been updated accordingly

Then fixed condor config vars: TRUST_DOMAIN is a condor expression, not a quoted string (suggested by TJ)

Added also unit tests for setup_x509.sh
And resed SEC_PASSWORD_DIRECTORY to an empty string (not needed in a client)

TODO: evaluate if the entry/credential trust domain from the Frontend configuration should be considered instead of the HTCSS one. Does it really matter what value is the TRUST_DOMAIN of the Glidein/startd? Will open an issue for it.

This includes and closes PR #223

### CODE REVIEW

File: creation/web_base/setup_x509.sh [hunk 1]
```diff
@@ -377,98 +386,118 @@ get_trust_domain() {
...
                 else
                     warn "$out"
                 fi
             else
+                # "$error_gen" -error "setup_x509.sh" "WN_Resource" "$STR"
                 warn "$out"
             fi
         else
-            # "$error_gen" -error "setup_x509.sh" "WN_Resource" "$STR"
+            #"$error_gen" -error "setup_x509.sh" "WN_Resource" "$STR1" "directory" "$X509_CERT_DIR"
```
Comment (namrathaurs) [inline creation/web_base/setup_x509.sh:485]: same comment about replacing the script name with `$0`

File: creation/web_base/setup_x509.sh [hunk 2]
```diff
@@ -377,98 +386,118 @@ get_trust_domain() {
...
-fi
 
-if [[ -z "$cred_updated" ]]; then
-    warn "setup_x509.sh: No valid credential (x509 proxy or idtoken) found"
-    "$error_gen" -error "setup_x509.sh" "WN_Resource" "No valid credential (x509 proxy or idtoken) found"
-    exit 1
-fi
+    if [[ -z "$cred_updated" ]]; then
+        warn "setup_x509.sh: No valid credential (x509 proxy or idtoken) found"
+        "$error_gen" -error "setup_x509.sh" "WN_Resource" "No valid credential (x509 proxy or idtoken) found"
```
Comment (namrathaurs) [inline creation/web_base/setup_x509.sh:492]: same comment as previous

File: creation/web_base/setup_x509.sh [hunk 3]
```diff
@@ -377,98 +386,118 @@ get_trust_domain() {
...
+        exit 1
+    fi
 
-warn "setup_x509.sh: Credentials copied, verified, amd added to glidein_config: ${cred_updated%,}"
-"$error_gen" -ok "setup_x509.sh" "idtoken" "${GLIDEIN_CONDOR_TOKEN:-unavailable}" "trust_domain" "${TRUST_DOMAIN:-unavailable}" "proxy" "${X509_USER_PROXY:-unavailable}" "cert_dir" "${X509_CERT_DIR:-unavailable}"
-exit 0
+    warn "setup_x509.sh: Credentials copied, verified, amd added to glidein_config: ${cred_updated%,}"
+    "$error_gen" -ok "setup_x509.sh" "idtoken" "${GLIDEIN_CONDOR_TOKEN:-unavailable}" "trust_domain" "${TRUST_DOMAIN:-unavailable}" "proxy" "${X509_USER_PROXY:-unavailable}" "cert_dir" "${X509_CERT_DIR:-unavailable}"
```
Comment (namrathaurs) [inline creation/web_base/setup_x509.sh:497]: same here

**Review by namrathaurs — CHANGES_REQUESTED** [review]
Very minor comments upon reviewing. Please check if they are valid, otherwise can be ignored.

File: creation/web_base/setup_x509.sh [hunk 4]
```diff
@@ -377,98 +386,118 @@ get_trust_domain() {
...
+        X509_USER_PROXY=$(gconfig_get X509_USER_PROXY "$glidein_config")
+        GLIDEIN_CONDOR_TOKEN=$(gconfig_get GLIDEIN_CONDOR_TOKEN "$glidein_config")
 
-if [[ -n "$GLIDEIN_PERIODIC_SCRIPT" ]]; then
-    X509_USER_PROXY_ORIG=$(gconfig_get X509_USER_PROXY_ORIG "$glidein_config")
-    GLIDEIN_CONDOR_TOKEN_ORIG=$(gconfig_get GLIDEIN_CONDOR_TOKEN_ORIG "$glidein_config")
-    X509_USER_PROXY=$(gconfig_get X509_USER_PROXY "$glidein_config")
-    GLIDEIN_CONDOR_TOKEN=$(gconfig_get GLIDEIN_CONDOR_TOKEN "$glidein_config")
+        if refresh_credentials "$X509_USER_PROXY_ORIG" "$X509_USER_PROXY" "$GLIDEIN_CONDOR_TOKEN_ORIG" "$GLIDEIN_CONDOR_TOKEN"; then
+            "$error_gen" -ok "setup_x509.sh" "idtoken" "${GLIDEIN_CONDOR_TOKEN:-unavailable}" "proxy" "${X509_USER_PROXY:-unavailable}"
```
[the remark this answers is not in the document]
Comment (mambelli) [reply creation/web_base/setup_x509.sh:422]: done

File: creation/web_base/setup_x509.sh [hunk 5]
```diff
@@ -377,98 +386,118 @@ get_trust_domain() {
 #
 ############################################################
 
-# Path read from and stored in glidein_config should always be full absolute paths
-GLIDEIN_CONDOR_TOKEN=$(gconfig_get GLIDEIN_CONDOR_TOKEN "$glidein_config")
-GLIDEIN_START_DIR_ORIG=$(gconfig_get GLIDEIN_START_DIR_ORIG "$glidein_config")
-GLIDEIN_WORKSPACE_ORIG=$(gconfig_get GLIDEIN_WORKSPACE_ORIG "$glidein_config")
+_main() {
+    # Aux scripts: import gconfig functions and define error_gen
+    add_config_line_source="$(grep '^ADD_CONFIG_LINE_SOURCE ' "$glidein_config" | cut -d ' ' -f 2-)"
```
[the remark this answers is not in the document]
Comment (mambelli) [reply creation/web_base/setup_x509.sh:391]: No, gconfig_... are defined only after add_config_line_source is retrieved and sourced

File: creation/web_base/setup_x509.sh [hunk 6]
```diff
@@ -377,98 +386,118 @@ get_trust_domain() {
...
-    GLIDEIN_CONDOR_TOKEN=$(gconfig_get GLIDEIN_CONDOR_TOKEN "$glidein_config")
+        if refresh_credentials "$X509_USER_PROXY_ORIG" "$X509_USER_PROXY" "$GLIDEIN_CONDOR_TOKEN_ORIG" "$GLIDEIN_CONDOR_TOKEN"; then
+            "$error_gen" -ok "setup_x509.sh" "idtoken" "${GLIDEIN_CONDOR_TOKEN:-unavailable}" "proxy" "${X509_USER_PROXY:-unavailable}"
+            exit 0
+        fi
 
-    if refresh_credentials "$X509_USER_PROXY_ORIG" "$X509_USER_PROXY" "$GLIDEIN_CONDOR_TOKEN_ORIG" "$GLIDEIN_CONDOR_TOKEN"; then
-        "$error_gen" -ok "setup_x509.sh" "idtoken" "${GLIDEIN_CONDOR_TOKEN:-unavailable}" "proxy" "${X509_USER_PROXY:-unavailable}"
-        exit 0
+        "$error_gen" -error "setup_x509.sh" "WN_Resource" "Periodic execution should not reach setup, probably needed glidein_config variables are not set"
```
[the remark this answers is not in the document]
Comment (mambelli) [reply creation/web_base/setup_x509.sh:426]: No, only if it fails, the if contains an `exit` command

File: creation/web_base/setup_x509.sh (hunk 1 above)
Comment (namrathaurs) [reply creation/web_base/setup_x509.sh:500]: The `setup_x509.sh` in this line might have been forgotten to rename the script to SCRIPT_NAME

**Review by namrathaurs — CHANGES_REQUESTED** [review]
Once the comment is addressed, this PR is ready for merging

**Comment by mambelli** [comment]
Those were in comments kept around for the error strings, anyway I removed them.

**Comment by namrathaurs** [comment]
My bad, did not notice closely that the line was commented out. It’s now ready for merging!

**Pull request merged**